docs/src/dev-guide/design-kv-ir-streams/background.md

```
@@ -0,0 +1,326 @@
+### Writing archives to disk
+
+To write an archive's data structures to disk, clp-s serializes them and writes them to one or more
+general-purpose compression streams. Applying general-purpose compression allows us to mitigate some
+of the inefficient encodings (e.g., encoding `Boolean`s as integers) used to maintain efficient
+search performance. For some data structures, like dictionaries, clp-s writes them to disk as they
+are built; yet for other data structures, like the ERTs, clp-s buffers them in memory until the
+archive is complete.
+
+## Parsing & encoding unstructured text
+
+clp-s uses [clp](../../user-guide/core-unstructured/clp.md)'s algorithm to parse and encode
```

markdownlint-cli2 (0.17.2)

Line length
Expected: 80; Actual: 91

(MD013, line-length)